diffrax/_misc.py

```diff
@@ -189,3 +195,20 @@ def upcast_or_raise(
     elif config_value != "standard":
         assert False, f"Unrecognised `JAX_NUMPY_DTYPE_PROMOTION={config_value}`"
     return jnp.astype(x, promote_dtype)
+
+
+def pending_deprecation_warning(cls):
+    orig_init = cls.__init__
+
+    @wraps(orig_init)
+    def new_init(self, *args, **kwargs):
+        warnings.warn(
+            f"{cls.__name__} is pending deprecation and may be removed "
+            "in future versions. Consider using the new alternative "
+            "ControlTerm(lx.DiagonalLinearOperator(...)).",
+            PendingDeprecationWarning,
+        )
+        orig_init(self, *args, **kwargs)
+
+    cls.__init__ = new_init
+    return cls
```

Comment:
Since I think we only use this once, let's inline it where it's used. (In particular as the error message mentions `ControlTerm(lx.DiagonalLinearOperator(...))`!)

Reply:
Ok, moved to be inside the weakly diagonal